security/tpm/test/_tpm_test.js

```diff
@@ -2,45 +2,151 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 'use strict';
-
+var common = require("azure-iot-common");
 var TpmSecurityClient  = require('../lib/tpm').TpmSecurityClient ;
+var tss = require("tss.js");
+var sinon = require('sinon');
+
 var assert = require('chai').assert;
 
-describe('tpm', function () {
-  this.timeout(1000);
+var fakeSimpleTpmClient = {
+  allowErrors: () => {return this},
+  connect: (callback) => {callback();},
+  ReadPublic: (handle, callback) => {
+    callback(null,null);
+  },
+  fakeSimpleTpmClient: () => {},
+  getLastResponseCode: () => {
+    return tss.TPM_RC.SUCCESS;
+  }
+
+}
+function testFalsyArg(methodUnderTest, argName, argValue, ExpectedErrorType) {
+  var errorName = ExpectedErrorType ? ExpectedErrorType.name : 'Error';
+  it('Throws a ' + errorName + ' if \'' + argName + '\' is \'' + JSON.stringify(argValue) + '\' (type:' + typeof(argValue) + ')', function() {
+    var client = new TpmSecurityClient(null, fakeSimpleTpmClient);
+    assert.throws(function() {
+      client[methodUnderTest](argValue, function() {});
+    }, ExpectedErrorType);
+  });
+}
+
 
-  var obj = new TpmSecurityClient ();
+describe('tpm', function () {
+  this.timeout(5000);
 
   describe('getEndorsementKey', function() {
-    it ('throws', function() {
-      assert.throws(function() {
-        obj.getEndorsementKey();
+    it('returns the endorsement key', function(done) {
+      /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_006: [The `getEndorsmentKey` function shall query the TPM hardware and return the `endorsementKey` in the callback.] */
+      var client = new TpmSecurityClient(undefined, fakeSimpleTpmClient );
+      var persistStub = sinon.stub(client,'_createPersistentPrimary');
```

Comment:
>_createPersistentPrimary [](start = 43, length = 24)

it looks like you're mocking some internal implementation details of the object under test instead of the dependency of the object. (the TSS.JS object itself). means you're just obfuscating a whole part of the code you're actually trying to test.